src/fat/direntry/ser_de.rs

```diff
@@ -46,20 +43,21 @@ pub(crate) struct LFNEntry {
 }
 
 impl LFNEntry {
-    pub(crate) fn get_byte_slice(&self) -> [u16; CHARS_PER_LFN_ENTRY] {
-        let mut slice = [0_u8; CHARS_PER_LFN_ENTRY * mem::size_of::<u16>()];
-
-        slice[..LFN_FIRST_CHARS * 2].copy_from_slice(&self.first_chars);
-        slice[LFN_FIRST_CHARS * 2..(LFN_FIRST_CHARS + LFN_MID_CHARS) * 2]
-            .copy_from_slice(&self.mid_chars);
-        slice[(LFN_FIRST_CHARS + LFN_MID_CHARS) * 2..].copy_from_slice(&self.last_chars);
-
-        let mut out_slice = [0_u16; CHARS_PER_LFN_ENTRY];
-        for (i, chunk) in slice.chunks(mem::size_of::<u16>()).enumerate() {
-            out_slice[i] = u16::from_le_bytes(chunk.try_into().unwrap());
+    pub(crate) fn copy_lfn_name(&self, slice: &mut [u16; CHARS_PER_LFN_ENTRY]) {
+        {
+            // SAFETY: The pointer below is properly aligned and we aren't accessing
+            // the original reference for as long as this pointer is in-scope
+            let slice = unsafe { &mut *(slice.as_mut_ptr() as *mut [u8; CHARS_PER_LFN_ENTRY * 2]) };
+
+            // copy the bytes from the lfn name into it
+            slice[..LFN_FIRST_CHARS * 2].copy_from_slice(&self.first_chars);
+            slice[LFN_FIRST_CHARS * 2..(LFN_FIRST_CHARS + LFN_MID_CHARS) * 2]
+                .copy_from_slice(&self.mid_chars);
+            slice[(LFN_FIRST_CHARS + LFN_MID_CHARS) * 2..].copy_from_slice(&self.last_chars);
         }
 
-        out_slice
+        // fix endian (if required)
+        slice.iter_mut().for_each(|c| *c = c.to_le());
```

Comment:
This bother me, since the LFN is little-endian by default. I will leave this but leave a comment saying that this probably needs to be looked into in the future

Reply:
I would hope that it's removed by the optimizer, or you can slap a `#[cfg(target_endian = "big")]` to it.